vumi/persist/riak_base.py

```diff
@@ -0,0 +1,208 @@
+"""Basic tools for building a Riak manager."""
+
+import json
+
+from riak import RiakClient
+
+
+def to_unicode(text, encoding='utf-8'):
+    if text is None:
+        return text
+    if isinstance(text, tuple):
+        return tuple(to_unicode(item, encoding) for item in text)
+    if not isinstance(text, unicode):
```

Comment:
The short and unsatisfactory answer is "I just moved the existing code without modifying it". The longer answer is that the current implementation probably had the tuple case added later and the structure made more sense before that.

I'll take a look at this and see if there's a better way to write this function, thanks.